pytorch_lightning/trainer/trainer.py

```diff
@@ -712,7 +712,10 @@ def run_evaluation(self, on_epoch=False):
         # update epoch-level lr_schedulers
         if on_epoch:
             self.optimizer_connector.update_learning_rates(interval='epoch')
-
+        elif not on_epoch and not self.sanity_checking:
```

Comment:
`interval='val'` won't be called `on_epoch=True`. I would assume it to be applied after each validation call.